src/testdrive/src/main.rs

```diff
@@ -200,8 +200,27 @@ async fn main() {
     let mut errors = Vec::new();
     let mut error_files = Vec::new();
 
-    if files.is_empty() {
-        files.push("-".to_string())
+    let mut files = vec![];
+
+    if args.globs.is_empty() {
+        files.push(PathBuf::from("-"))
+    } else {
+        let walker = match GlobWalkerBuilder::from_patterns(".", &args.globs).build() {
+            Ok(walker) => walker,
+            Err(e) => {
+                eprintln!("unable to construct glob matcher: {}", e);
+                process::exit(1);
+            }
+        };
+        for dir_entry in walker {
+            match dir_entry.err_ctx("walking directory") {
+                Ok(dir_entry) => files.push(dir_entry.path().to_path_buf()),
+                Err(e) => {
+                    eprintln!("{}", e);
+                    errors.push(e);
+                }
+            }
+        }
```

Comment:
Now that we have control over globbing I think this should error out if the glob failed to match anything. Otherwise we risk test bugs where testdrive does not run anything without us noticing. Previously, failing to provide any .td files caused testdrive to hang waiting for input from STDIN, which was noticeable.